pylxd/models/operation.py

```diff
@@ -64,3 +84,15 @@ def wait(self):
         except KeyError:
             # Support for legacy LXD
             pass
+
+    def cancel(self):
+        """
+        Cancel an operation. If may_cancel is False do nothing.
+
+        Calling this will change the state to "cancelling" rather than
+        actually removing the entry.
+        """
+        if not self.may_cancel:
+            return
+        self.api.delete()
+        self.status = "cancelling"
```

Comment:
We shouldn't set `self.status = "cancelling"` as it might not reflect the corresponding state in the object in lxd.
What we should do here is fetch the operation again, and copy the `status`, `status_code`, `may_cancel` and `err` attributes back onto the class.

Alternatively, we should promote it to a `_Model` and just call `sync()` which might be simpler, overall.